Gems/Terrain/Code/Source/TerrainSystem/TerrainSystem.h

```diff
@@ -0,0 +1,129 @@
+        //!                  otherwise *terrainExistsPtr will become true.
+        float GetHeight(AZ::Vector3 position, Sampler sampler = Sampler::BILINEAR, bool* terrainExistsPtr = nullptr) const override;
+        float GetHeightFromFloats(float x, float y, Sampler sampler = Sampler::BILINEAR, bool* terrainExistsPtr = nullptr) const override;
+
+        //! Given an XY coordinate, return the max surface type and weight.
+        //! @terrainExists: Can be nullptr. If != nullptr then, if there's no terrain at location x,y or location x,y is inside a terrain
+        //! HOLE then *terrainExistsPtr will be set to false,
+        //!                  otherwise *terrainExistsPtr will be set to true.
+        AzFramework::SurfaceData::SurfaceTagWeight GetMaxSurfaceWeight(
+            AZ::Vector3 position, Sampler sampleFilter = Sampler::BILINEAR, bool* terrainExistsPtr = nullptr) const override;
+        AzFramework::SurfaceData::SurfaceTagWeight GetMaxSurfaceWeightFromFloats(
+            float x, float y, Sampler sampleFilter = Sampler::BILINEAR, bool* terrainExistsPtr = nullptr) const override;
+
+        //! Convenience function for  low level systems that can't do a reverse lookup from Crc to string. Everyone else should use
+        //! GetMaxSurfaceWeight or GetMaxSurfaceWeightFromFloats. Not available in the behavior context. Returns nullptr if the position is
+        //! inside a hole or outside of the terrain boundaries.
+        const char* GetMaxSurfaceName(
+            AZ::Vector3 position, Sampler sampleFilter = Sampler::BILINEAR, bool* terrainExistsPtr = nullptr) const override;
+
+        //! Returns true if there's a hole at location x,y.
+        //! Also returns true if there's no terrain data at location x,y.
+        bool GetIsHoleFromFloats(float x, float y, Sampler sampleFilter = Sampler::BILINEAR) const override;
+
+        // Given an XY coordinate, return the surface normal.
+        //! @terrainExists: Can be nullptr. If != nullptr then, if there's no terrain at location x,y or location x,y is inside a terrain
+        //! HOLE then *terrainExistsPtr will be set to false,
+        //!                  otherwise *terrainExistsPtr will be set to true.
+        AZ::Vector3 GetNormal(
+            AZ::Vector3 position, Sampler sampleFilter = Sampler::BILINEAR, bool* terrainExistsPtr = nullptr) const override;
+        AZ::Vector3 GetNormalFromFloats(
+            float x, float y, Sampler sampleFilter = Sampler::BILINEAR, bool* terrainExistsPtr = nullptr) const override;
+
+    private:
+        float GetHeightSynchronous(float x, float y) const;
+        AZ::Vector3 GetNormalSynchronous(float x, float y) const;
+
+        /*
+        void GetSurfaceWeights(const AZ::Vector3& inPosition, Sampler sampleFilter, SurfaceData::SurfaceTagWeightMap& outSurfaceWeights)
+        override; void GetSurfacePoint(const AZ::Vector3& inPosition, Sampler sampleFilter, SurfaceData::SurfacePoint& outSurfacePoint)
```

Comment:
I know this is just here for posterity, but it looks like the declarations are on mixed lines?

Reply:
Whoa, that went weird!  I'll just delete these lines for now.